lib/MySQL_Session.cpp

```diff
@@ -2433,22 +2428,21 @@ bool MySQL_Session::handler_again___status_SETTING_MULTI_STMT(int *_rc) {
 		if (rc==-1) {
 			// the command failed
 			int myerr=mysql_errno(myconn->mysql);
-			// if rc was '-1' but 'mysql_errno' is 0, it means that the connection
-			// wasn't stablished because the server was detected to be down, *prior*
-			// to the connection attempt, and the session should be destroyed.
 			if (myerr >= 2000 || myerr == 0) {
 				bool retry_conn=false;
 				// client error, serious
 				if (myerr != 0) {
 					proxy_error("Detected a broken connection during setting MYSQL_OPTION_MULTI_STATEMENTS on %s , %d : %d, %s\n", myconn->parent->address, myconn->parent->port, myerr, mysql_error(myconn->mysql));
+					MyHGM->p_update_mysql_error_counter(p_mysql_error_type::mysql, myconn->parent->myhgc->hid, myconn->parent->address, myconn->parent->port, myerr);
 				} else {
 					proxy_error(
 						"Detected a broken connection during setting MYSQL_OPTION_MULTI_STATEMENTS on %s , %d : %d, %s\n",
 						myconn->parent->address,
 						myconn->parent->port,
-						myerr,
+						ER_PROXYSQL_OFFLINE_SRV,
```

Comment:
This can stay `myerr` .
For the same replace didn't happen in othe functions, like `handler_again___status_SETTING_INIT_CONNECT`